chunky/src/java/se/llbit/chunky/renderer/scene/PathTracer.java

```diff
@@ -507,11 +532,12 @@ private static void sampleEmitterFace(Scene scene, Ray ray, Grid.EmitterPosition
         double e = Math.abs(emitterRay.d.dot(emitterRay.getNormal()));
         e /= Math.max(distance * distance, 1);
         e *= pos.block.surfaceArea(face);
-        e *= emitterRay.getCurrentMaterial().emittance;
-        e *= scene.emitterIntensity;
         e *= scaler;
+        Vector3 emittance = new Vector3();
+        doEmittanceMapping(emittance, emitterRay.color, scene, emitterRay.getCurrentMaterial());
+        emittance.scale(e);
 
-        result.scaleAdd(e, emitterRay.color);
+        result.add(new Vector4(emittance, 0));
```

Comment:
Can we do this without allocating `Vector3` and `Vector4`?

Reply:
I think the `Vector3 emiitance` is definitely useful because otherwise we'd need a `doEmittanceMappingR` etc. but the `Vector4` could be removed